pkg/ingestor/puller/blob/blob_test.go

```diff
@@ -0,0 +1,525 @@
+					},
+				},
+			},
+			want: &BlobStore{
+				bucketName: "fakeBlobStorage",
+				cfg: &config.KubehoundConfig{
+					Ingestor: config.IngestorConfig{
+						TempDir: getTempDir(t),
+					},
+				},
+				region: "",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			got, err := NewBlobStorage(tt.args.cfg, tt.args.blobConfig)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewBlobStorage() error = %v, wantErr %v", err, tt.wantErr)
+
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewBlobStorage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestBlobStore_Put(t *testing.T) {
+	t.Parallel()
+	fakeBlobStoragePut := "./testdata/fakeBlobStoragePut"
+	bucketNamePut := fmt.Sprintf("file://%s", fakeBlobStoragePut)
+	type fields struct {
+		bucketName string
+		cfg        *config.KubehoundConfig
+		region     string
+	}
+	type args struct {
+		archivePath string
+		clusterName string
+		runID       string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Push new dump",
+			fields: fields{
+				bucketName: bucketNamePut,
+				cfg:        dummyKubehoundConfig(t),
+			},
+			args: args{
+				archivePath: "./testdata/archive.tar.gz",
+				clusterName: validClusterName,
+				runID:       "91j2qs8th6yarr5hkafysekn0j",
+			},
+			wantErr: false,
+		},
+		{
+			name: "non existing filepath",
+			fields: fields{
+				bucketName: bucketNamePut,
+				cfg:        dummyKubehoundConfig(t),
+			},
+			args: args{
+				archivePath: "./testdata/archive2.tar.gz",
+				clusterName: validClusterName,
+				runID:       "91j2qs8th6yarr5hkafysekn0j",
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid runID",
+			fields: fields{
+				bucketName: bucketNamePut,
+				cfg:        dummyKubehoundConfig(t),
+			},
+			args: args{
+				archivePath: "./testdata/archive2.tar.gz",
+				clusterName: validClusterName,
+				runID:       "91j2qs8th6yarr5hkafysekn0T",
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ctx := context.Background()
+			bs := &BlobStore{
+				bucketName: tt.fields.bucketName,
+				cfg:        tt.fields.cfg,
+				region:     tt.fields.region,
+			}
+			var err error
+			if err = bs.Put(ctx, tt.args.archivePath, tt.args.clusterName, tt.args.runID); (err != nil) != tt.wantErr {
+				t.Errorf("BlobStore.Put() error = %v, wantErr %v", err, tt.wantErr)
+			}
+
+			if err != nil {
+				return
+			}
+
+			// Building result path to clean up
+			dumpResult, err := dump.NewDumpResult(tt.args.clusterName, tt.args.runID, true)
+			if err != nil {
+				t.Errorf("NewDumpResult cluster:%s runID:%s", tt.args.clusterName, tt.args.runID)
+			}
+			key := path.Join(bs.cfg.Ingestor.TempDir, dumpResult.GetFullPath())
+
+			err = os.RemoveAll(path.Join(fakeBlobStoragePut, tt.args.clusterName))
```

Comment:
That sounds like a dangerous operation to run on a test 😅

Reply:
Yes I know, but otherwise, it will let dirty files locally.